test/integration/controllermanager/managedseedset/managedseedset_test.go

```diff
@@ -0,0 +1,367 @@
+					ObjectMeta: managedSeed.ObjectMeta,
+					Spec:       managedSeed.Spec,
+				},
+				ShootTemplate: gardencorev1beta1.ShootTemplate{
+					ObjectMeta: shoot.ObjectMeta,
+					Spec:       shoot.Spec,
+				},
+				UpdateStrategy: &seedmanagementv1alpha1.UpdateStrategy{
+					Type: updateStrategyTypePtr(seedmanagementv1alpha1.RollingUpdateStrategyType),
+					RollingUpdate: &seedmanagementv1alpha1.RollingUpdateStrategy{
+						Partition: pointer.Int32(0),
+					},
+				},
+			},
+		}
+
+		shootList = &gardencorev1beta1.ShootList{}
+		managedSeedList = &seedmanagementv1alpha1.ManagedSeedList{}
+		selector, err = metav1.LabelSelectorAsSelector(&managedSeedSet.Spec.Selector)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	JustBeforeEach(func() {
+		By("Create ManagedSeedSet")
+		Expect(testClient.Create(ctx, managedSeedSet)).To(Succeed())
+		log.Info("Created ManagedSeedSet for test", "managedSeedSet", client.ObjectKeyFromObject(managedSeedSet))
+
+		DeferCleanup(func() {
+			By("Delete ManagedSeedSet")
+			Expect(testClient.Delete(ctx, managedSeedSet)).To(Or(Succeed(), BeNotFoundError()))
+			Eventually(func() error {
+				return mgrClient.Get(ctx, client.ObjectKeyFromObject(managedSeedSet), managedSeedSet)
+			}).Should(BeNotFoundError())
+		})
+	})
+
+	Context("reconcile", func() {
+		It("should add the finalizer", func() {
+			Eventually(func(g Gomega) {
+				g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(managedSeedSet), managedSeedSet)).To(Succeed())
+				g.Expect(managedSeedSet.Finalizers).To(ConsistOf("gardener"))
+			}).Should(Succeed())
+		})
+
+		It("should create Shoot from shoot template and set the status.replica value to 1 (default value)", func() {
+			Eventually(func(g Gomega) {
+				g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(managedSeedSet), managedSeedSet)).To(Succeed())
+				g.Expect(managedSeedSet.Status.Replicas).To(Equal(int32(1)))
+				g.Expect(testClient.List(ctx, shootList, client.InNamespace(managedSeedSet.Namespace), client.MatchingLabelsSelector{Selector: selector})).To(Succeed())
+				g.Expect(shootList.Items).To(HaveLen(1))
+			}).Should(Succeed())
+		})
+
+		It("should create ManagedSeed when shoot is reconciled successfully", func() {
+			makeShootStateSucceeded()
+
+			Eventually(func(g Gomega) {
+				g.Expect(testClient.List(ctx, managedSeedList, client.InNamespace(managedSeedSet.Namespace), client.MatchingLabelsSelector{Selector: selector})).To(Succeed())
+				g.Expect(managedSeedList.Items).To(HaveLen(1))
+				g.Expect(managedSeedList.Items[0].Spec.Shoot.Name).To(Equal(shootList.Items[0].Name))
+				g.Expect(managedSeedList.Items[0].Spec.SeedTemplate.Spec.Provider.Type).To(Equal("providerType"))
+				g.Expect(managedSeedList.Items[0].Spec.SeedTemplate.Spec.DNS.IngressDomain).To(Equal(seed.Spec.DNS.IngressDomain))
+				g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(managedSeedSet), managedSeedSet)).To(Succeed())
+				g.Expect(managedSeedSet.Status.PendingReplica).NotTo(BeNil())
```

Comment:
Instead of checking `NotTo(BeNil())`, shouldn't we check `Equal(pointer.Int32(1))`?

Reply:
No, pendingReplica is not an integer value https://github.com/gardener/gardener/blob/40b0bf10522e53c55ceaf1b361a4d016a94fe741/pkg/apis/seedmanagement/v1alpha1/types_managedseedset.go#L143

Reply:
Ah, OK, I wasn't aware. Thanks :) Then disregard these comments.